### DISCUSSION TITLE
Is there a way to auto-materialize assets in dagster cloud after a production deployment?

### DISCUSSION THREAD

**Discussion by gnilrets** [dagster-io/dagster#17906] [Q&A]
I have some assets that only need to be materialized when code is deployed.  I'm thinking I could probably set up a job that runs those assets and then trigger the job in CI after deployment, but I was wondering if there was some other way to do it with auto-materialization?

**Comment by OwenKephart** [comment]
Currently, this is not possible via auto-materialization, although one direction we've considered is along the lines of correlating deployment with code version changes, and then implementing a rule along the lines of this issue: https://github.com/dagster-io/dagster/issues/15686.

Would this sort of scheme work for your use case, or are you looking to update your assets regardless of if the underlying code has changed?

**Reply by gnilrets** [reply]
I think I'm probably not interested in using asset code versions right now.  Seems like it would be too easy to actually make a change and forget to manually bump the version.

There's some situations where I'd only want to update assets when the underlying code changes for efficiency purposes, but for now I'm ok updating them every time time a code change _potentially_ could have happened (e.g., deployment).

**Reply by OwenKephart** [reply]
Got it, that makes sense -- that specific use case is not on the short-term roadmap (and so instead I'd recommend the approach that you mentioned originally of kicking off a job from CI)

In the more medium-long term, we're imagining a world in which custom code can be incorporated into the auto-materialize system, so conceivably that code could be responsible for detecting code version changes.

However, it's not clear that this would be any simpler / better than just kicking off a job from CI.
